tendrl/node_agent/sds_detection/detect_gluster_sds_details.py

```diff
@@ -0,0 +1,50 @@
+import hashlib
+import logging
+import subprocess
+
+from tendrl.node_agent.auto_detect.sds_detect_plugin import SDSDetectPlugin
+
+LOG = logging.getLogger(__name__)
+
+
+class DetectGlusterStorageSystem(SDSDetectPlugin):
+    def detect_storage_system(self):
+        ret_val = {}
+
+        # get the gluster version details
+        cmd = subprocess.Popen(
+            "gluster --version",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        out, err = cmd.communicate()
+        if err and 'command not found' in err:
+            LOG.error("Error getting gluster version details")
+            return ret_val
+        lines = out.split('\n')
+        ret_val['pkg_version'] = lines[0].split()[1]
+        ret_val['pkg_name'] = lines[0].split()[0]
+        ret_val['roles'] = ['glusterd']
+
+        # form the temporary cluster_id
+        cmd = subprocess.Popen(
+            "gluster pool list",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        out, err = cmd.communicate()
+        if err:
+            LOG.error("Error formulating cluster_id")
+            return ret_val
+        lines = out.split('\n')[1:]
```

Comment:
thats a good point. I feel it will differ. will need to sort the host list and then calculate the checksum I feel. Will do that.